test/algorithms/DominguezRios.jl

```diff
@@ -0,0 +1,513 @@
+        -2092 -3244 -2643 -2705
+        -2146 -1944 -2947 -3428
+        -2298 -4034 -1885 -1885
+    ]
+    N = MOI.get(model, MOI.ResultCount())
+    x_sol = hcat([MOI.get(model, MOI.VariablePrimal(i), x) for i in 1:N]...)
+    @test isapprox(x_sol, X_E'; atol = 1e-6)
+    y_sol = hcat([MOI.get(model, MOI.ObjectiveValue(i)) for i in 1:N]...)
+    @test isapprox(y_sol, Y_N'; atol = 1e-6)
+    return
+end
+
+function test_knapsack_max_p4()
+    p = 4
+    n = 10
+    W = 2653.0
+    C = Float64[
+        566 611 506 180 817 184 585 423 26 317
+        62 84 977 979 874 54 269 93 881 563
+        664 982 962 140 224 215 12 869 332 537
+        557 898 148 63 78 964 246 662 386 272
+    ]
+    w = Float64[979 448 355 955 426 229 9 695 322 889]
+    model = MOA.Optimizer(HiGHS.Optimizer)
+    MOI.set(model, MOA.Algorithm(), MOA.DominguezRios())
+    MOI.set(model, MOI.Silent(), true)
+    x = MOI.add_variables(model, n)
+    MOI.add_constraint.(model, x, MOI.ZeroOne())
+    MOI.add_constraint(
+        model,
+        MOI.ScalarAffineFunction(
+            [MOI.ScalarAffineTerm(w[j], x[j]) for j in 1:n],
+            0.0,
+        ),
+        MOI.LessThan(W),
+    )
+    f = MOI.VectorAffineFunction(
+        [
+            MOI.VectorAffineTerm(i, MOI.ScalarAffineTerm(C[i, j], x[j])) for
+            i in 1:p for j in 1:n
+        ],
+        fill(0.0, p),
+    )
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+    MOI.set(model, MOI.ObjectiveFunction{typeof(f)}(), f)
+    MOI.optimize!(model)
+    X_E = Float64[
+        0 1 1 0 1 1 1 1 1 0
+        0 1 1 1 1 0 1 0 1 0
+        0 1 1 0 1 0 1 0 1 1
+        0 0 1 0 1 1 1 0 1 1
+        1 1 1 0 1 1 1 0 0 0
+        0 0 1 1 0 1 1 1 1 0
+        0 1 1 1 1 1 1 0 0 0
+        0 1 1 1 0 1 1 0 1 0
+        0 1 0 0 0 1 1 1 1 1
+        0 0 1 1 1 1 1 0 1 0
+    ]
+    Y_N = Float64[
+        3152 3232 3596 3382
+        2725 4064 2652 1819
+        2862 3648 3049 2028
+        2435 3618 2282 2094
+        3269 2320 3059 2891
+        1904 3253 2530 2469
+        2883 3237 2535 2397
+        2092 3244 2643 2705
+        2146 1944 2947 3428
+        2298 4034 1885 1885
+    ]
+    N = MOI.get(model, MOI.ResultCount())
+    x_sol = hcat([MOI.get(model, MOI.VariablePrimal(i), x) for i in 1:N]...)
+    @test isapprox(x_sol, X_E'; atol = 1e-6)
+    y_sol = hcat([MOI.get(model, MOI.ObjectiveValue(i)) for i in 1:N]...)
+    @test isapprox(y_sol, Y_N'; atol = 1e-6)
+    return
+end
+
+function test_assignment_min_p3()
+    p = 3
+    n = 5
+    C = Float64[
+        6 1 20 2 3
+        2 6 9 10 18
+        1 6 20 5 9
+        6 8 6 9 6
+        7 10 10 6 2
+        17 20 8 8 20
+        10 13 1 10 15
+        4 11 1 13 1
+        19 13 7 18 17
+        15 3 5 1 11
+        10 7 1 19 12
+        2 15 12 10 3
+        11 20 16 12 9
+        10 15 20 11 7
+        1 9 20 7 6
+    ]
+    C = permutedims(reshape(C, (n, p, n)), [2, 1, 3])
+    model = MultiObjectiveAlgorithms.Optimizer(HiGHS.Optimizer)
```

Comment:
See all the changes I made to https://github.com/jump-dev/MultiObjectiveAlgorithms.jl/pull/22